core/src/km_core_processevent_api.cpp

```diff
@@ -50,7 +50,11 @@ km_core_process_event(km_core_state *state,
   if(state == nullptr) {
     return KM_CORE_STATUS_INVALID_ARGUMENT;
   }
-  return state->processor().process_event(state, vk, modifier_state, is_key_down, event_flags);
+  km_core_status status = state->processor().process_event(state, vk, modifier_state, is_key_down, event_flags);
+
+  state->prepare_actions();
```

Comment:
This both updates the actions struct and brings app_context back into sync with cached context. Perhaps could benefit from a rename?

Reply:
I suggest `apply_actions`.

Reply:
```suggestion
  state->apply_actions_and_merge_app_context();
```